api/controllers/MerchStoreController.ts

```diff
@@ -325,6 +319,14 @@ export class MerchStoreController {
     return { error: null };
   }
 
+  @Post('/order/pickup/:uuid/complete')
+  async completePickupEvent(@Params() params: UuidParam, @AuthenticatedUser() user: UserModel):
+  Promise<CompleteOrderPickupEventResponse> {
+    if (!PermissionsService.canManagePickupEvents(user)) throw new ForbiddenError();
+    const ordersMarkedAsMissed = await this.merchStoreService.completeOrderPickupEvent(params.uuid);
```

Comment:
For readability and style, would we want the controller to have the higher level idea completeing a pickup event while the service would have a more low level specific function (i.e. the service would just have a funciton "markMissedPickUps") or is that not necessary/more confusing

Reply:
Good question - with controllers, their main role is to take an incoming request, validate it, send it to the service for executing some business logic that generates a response, then return that response to the client. In the case you mentioned, 'completing a pickup event' constitutes business logic, so the service would handle that.  The repository layer is in charge of doing any database-level operations, and does not need to know any business logic (e.g. instead of it having a `markMissedPickUps` function, it can have a `updateOrderStatus(status: OrderStatus)` function). Hopefully the responsibilities of each layer is a bit clearer

Reply:
Ahhhh, Okay I see my misconception - in that case, this makes more sense 👍